### ISSUE TITLE
Use the new zstd compression of kernel 5.9

### ISSUE THREAD

**Issue by iblech** [NixOS/nixpkgs#101108]
The latest major version 5.9 of the Linux kernel [brings us support for a zstd-compressed kernel and initrd image](https://kernelnewbies.org/Linux_5.9#Support_for_ZSTD_compressed_kernel.2C_ramdisk_and_initramfs). Apparently, zstd compressed things decompress much faster and can cut boot times substantially, hence it seems worthwhile for us to support this scheme. Currently we are using bzip2.

Unfortunately, I probably won't find time to work on this myself, but thought to open this issue to track progress.

**Comment by eadwu** [comment]
I've [probably] been using it through a xanmod patch, not recently since the 5.9 release though. What I have in my files are

```
        # Kernel compression algorithm
        KERNEL_XZ = mkForce no;
        KERNEL_LZ4 = mkForce no;
        KERNEL_ZSTD = mkForce yes;
```

and adding `zstd` to `nativeBuildInputs` of the kernel derivation

**Comment by S-NA** [comment]
For the initrd image:

For me (which might differ from the stock kernel NixOS provides):
```
$ zgrep CONFIG_RD /proc/config.gz
# CONFIG_RD_GZIP is not set
# CONFIG_RD_BZIP2 is not set
# CONFIG_RD_LZMA is not set
# CONFIG_RD_XZ is not set
CONFIG_RD_LZO=y
# CONFIG_RD_LZ4 is not set
CONFIG_RD_ZSTD=y
```

`CONFIG_RD_ZSTD` is already enabled, so set `boot.initrd.compressor = "${lib.getBin pkgs.zstd}/bin/zstd";`
This enables ZSTD compression for initrd.

However, `# CONFIG_KERNEL_ZSTD is not set` which will need `zstd` added to `nativeBuildInputs` as mentioned above.

**Comment by archseer** [comment]
I was wondering if zstd should be made a part of stdenv. `gzip bzip2 xz zlib` are all available in stdenv on linux and with `zstd` slowly becoming more common by being available in the kernel and used by various distribution's packaging it should make sense to include it.

**Comment by bbigras** [comment]
Can this be closed? Did #107703 cover it all?

**Comment by iblech** [comment]
Yes, it did! Closing this issue. Thank you for pointing that out, didn't notice before.
